ts/components/session/menu/Menu.tsx

```diff
@@ -125,6 +122,31 @@ export function getInviteContactMenuItem(
   return null;
 }
 
+export interface PinConversationMenuItemProps {
+  conversationId: string;
+}
+
+export const MenuItemPinConversation = (
+  props: PinConversationMenuItemProps
+): JSX.Element | null => {
+  const { conversationId } = props;
+  const conversation = getConversationController()
+    .get(conversationId)
+    .getProps();
+  const { isPinned } = conversation;
+
+  const togglePinConversation = () => {
+    window.inboxStore?.dispatch(
+      conversationActions.conversationChanged(conversationId, {
+        ...conversation,
+        isPinned: !isPinned,
+      })
```

Comment:
that would be conversation.set('isPinned', !isPinned); await conversation.commit()